core/modules/memman/MemFileSet.cc

```diff
@@ -0,0 +1,139 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "memman/MemFileSet.h"
+
+// System Headers
+#include <errno.h>
+
+// Qserv Headers
+#include "memman/MemFile.h"
+#include "memman/Memory.h"
+
+namespace lsst {
+namespace qserv {
+namespace memman {
+
+/******************************************************************************/
+/*                            D e s t r u c t o r                             */
+/******************************************************************************/
+  
+MemFileSet::~MemFileSet() {
+
+    // Unreference every fle in our file set. This action will also cause
+    // memory to be unlocked if no one else is using the file then the file
+    // object will be deleted as well.
+    //
+    for (auto mfP : _lockFiles) {mfP->release();}
+    for (auto mfP : _flexFiles) {mfP->release();}
+}
+
+/******************************************************************************/
+/*                                   a d d                                    */
+/******************************************************************************/
+
+int MemFileSet::add(std::string const& tabname, int chunk,
+                    bool iFile, bool mustLK) {
+
+    std::string fPath(_memory.filePath(tabname, chunk, iFile));
+
+    // Obtain a memory file object for this table and chunk
+    //
+    MemFile::MFResult mfResult = MemFile::obtain(fPath, _memory, !mustLK);
+    if (mfResult.mfP == 0) return mfResult.retc;
+
+    // Add to the appropriate file set
+    //
+    if (mustLK) {
+        _lockFiles.push_back(mfResult.mfP);
+    } else {
+        _flexFiles.push_back(mfResult.mfP);
+    }
+    _numFiles++;
+    return 0;
+}
+  
+/******************************************************************************/
+/*                               l o c k A l l                                */
+/******************************************************************************/
+
+int MemFileSet::lockAll() {
+
+    MemFile::MLResult mlResult;
+    uint64_t bytesLocked, bytesMax, freeBytes;
+
+    // Calculate the number of bytes available at this point. Note that we force
+    // freeBytes to be atleast 1 to make memlock check before it tries locking
+    // the file to avoid a useless memory map operation if it can't be locked.
+    // By the time we get here someone else may have already locked the file.
+    //
+    bytesMax    = _memory.bytesMax();
+    bytesLocked = _memory.bytesLocked();
+    if (bytesMax <= bytesLocked) freeBytes = 1;
```

Comment:
hmm, you use fully line broken if/else above (line 68) but non-broken if/else here and below in the same file?  pick consistent style throughout file?